modules/eks/submodules/privatelink/variables.tf

```diff
@@ -0,0 +1,97 @@
+variable "deploy_id" {
+  type        = string
+  description = "Domino Deployment ID"
+
+  validation {
+    condition     = can(regex("^[a-z-0-9]{3,32}$", var.deploy_id))
+    error_message = "Argument deploy_id must: start with a letter, contain lowercase alphanumeric characters(can contain hyphens[-]) with length between 3 and 32 characters."
+  }
+}
+
+variable "region" {
+  type        = string
+  description = "AWS region for the deployment"
+  nullable    = false
+  validation {
+    condition     = can(regex("(us(-gov)?|ap|ca|cn|eu|sa|me|af|il)-(central|(north|south)?(east|west)?)-[0-9]", var.region))
+    error_message = "The provided region must follow the format of AWS region names, e.g., us-west-2, us-gov-west-1."
+  }
+}
+
+variable "vpc_endpoint_services" {
+  description = <<EOF
+    [{
+      name      = Name of the VPC Endpoint Service.
+      ports     = List of ports exposing the VPC Endpoint Service. i.e [8080, 8081]
+      cert_arn  = Certificate ARN used by the NLB associated for the given VPC Endpoint Service.
+      private_dns = Private DNS for the VPC Endpoint Service.
+    }]
+  EOF
+
+  type = list(object({
+    name        = optional(string)
+    ports       = optional(list(number))
+    cert_arn    = optional(string)
+    private_dns = optional(string)
+  }))
+}
+
+variable "route53_hosted_zone_name" {
```

Comment:
Given that this variable is `required` why are we toggling later on?
```
var.route53_hosted_zone_name != null ? 1 : 0
```

Reply:
it is required for the private link module, but optional for the eks, it could receive null

Reply:
fixing moving to object variable with validation